### ISSUE TITLE
What is the maintenance status of this package?

### ISSUE THREAD

**Issue by LemonBreezes** [remyhonig/elfeed-org#82]
Hi. I tried to use this package a few days ago and noticed it was completely broken. There was a missing function definition, an infinite loop caused by a mis-named variable, among other things. I have completely fixed this package back to its former glory in https://github.com/remyhonig/elfeed-org/pull/81.

**Comment by mpsda** [comment]
Hi @LemonBreezes, thanks for fixing this package back to its former glory! After #81 was merged I'm getting an error for `elfeed-org-export-opml` function. Have you experienced this error as well?

Backtrace:

```elisp
Debugger entered--Lisp error: (wrong-type-argument fixnump t)
  string-match("^\\[\\[\\(http.+?\\)\\]\\[\\(.+?\\)\\]\\]" "root" t)
  #f(compiled-function (h) #<bytecode 0x10fd8ec9c4b56a3b>)((headline ... ... ... ... ... ... ...))
  #f(compiled-function (--data) #<bytecode -0x14bda4bc676cbc52>)((headline ... ... ... ... ... ... ...))
  mapc(#f(compiled-function (--data) #<bytecode -0x14bda4bc676cbc52>) (...))
  #f(compiled-function (--data) #<bytecode -0x14bda4bc676cbc52>)((...))
  org-element-ast-map((...) headline #f(compiled-function (h) #<bytecode 0x10fd8ec9c4b56a3b>) nil nil (bold citation citation-reference code entity export-snippet footnote-reference inline-babel-call inline-src-block italic line-break latex-fragment link macro ...) nil t nil)
  org-element-map((...) headline #f(compiled-function (h) #<bytecode 0x10fd8ec9c4b56a3b>))
  rmh-elfeed-org-convert-org-to-opml(#<buffer elfeed.org>)
  #<subr elfeed-org-export-opml>()
  apply(#<subr elfeed-org-export-opml> nil)
  elfeed-org-export-opml()
  #<subr funcall-interactively>(elfeed-org-export-opml)
  apply(#<subr funcall-interactively> elfeed-org-export-opml)
  funcall-interactively(elfeed-org-export-opml)
  #<subr call-interactively>(elfeed-org-export-opml record nil)
  apply(#<subr call-interactively> (elfeed-org-export-opml record nil))
  explain-pause--wrap-call-interactively(#<subr call-interactively> elfeed-org-export-opml record nil)
  apply(explain-pause--wrap-call-interactively #<subr call-interactively> (elfeed-org-export-opml record nil))
  call-interactively(elfeed-org-export-opml record nil)
  command-execute(elfeed-org-export-opml record)
  execute-extended-command(nil "elfeed-org-export-opml" nil)
  #<subr funcall-interactively>(execute-extended-command nil "elfeed-org-export-opml" nil)
  apply(#<subr funcall-interactively> (execute-extended-command nil "elfeed-org-export-opml" nil))
  funcall-interactively(execute-extended-command nil "elfeed-org-export-opml" nil)
  #<subr call-interactively>(execute-extended-command nil nil)
  apply(#<subr call-interactively> (execute-extended-command nil nil))
  explain-pause--wrap-call-interactively(#<subr call-interactively> execute-extended-command nil nil)
  apply(explain-pause--wrap-call-interactively #<subr call-interactively> (execute-extended-command nil nil))
  call-interactively(execute-extended-command nil nil)
  command-execute(execute-extended-command)
```

**Comment by CrCs2O4** [comment]
Hi @LemonBreezes, thanks for fixing this package back to its former glory! After #81 was merged I'm getting an error for `elfeed-org-export-opml` function. Have you experienced this error as well?

Backtrace:

```elisp
Debugger entered--Lisp error: (wrong-type-argument fixnump t)
  string-match("^\\[\\[\\(http.+?\\)\\]\\[\\(.+?\\)\\]\\]" "root" t)
  #f(compiled-function (h) #<bytecode 0x10fd8ec9c4b56a3b>)((headline ... ... ... ... ... ... ...))
  #f(compiled-function (--data) #<bytecode -0x14bda4bc676cbc52>)((headline ... ... ... ... ... ... ...))
  mapc(#f(compiled-function (--data) #<bytecode -0x14bda4bc676cbc52>) (...))
  #f(compiled-function (--data) #<bytecode -0x14bda4bc676cbc52>)((...))
  org-element-ast-map((...) headline #f(compiled-function (h) #<bytecode 0x10fd8ec9c4b56a3b>) nil nil (bold citation citation-reference code entity export-snippet footnote-reference inline-babel-call inline-src-block italic line-break latex-fragment link macro ...) nil t nil)
  org-element-map((...) headline #f(compiled-function (h) #<bytecode 0x10fd8ec9c4b56a3b>))
  rmh-elfeed-org-convert-org-to-opml(#<buffer elfeed.org>)
  #<subr elfeed-org-export-opml>()
  apply(#<subr elfeed-org-export-opml> nil)
  elfeed-org-export-opml()
  #<subr funcall-interactively>(elfeed-org-export-opml)
  apply(#<subr funcall-interactively> elfeed-org-export-opml)
  funcall-interactively(elfeed-org-export-opml)
  #<subr call-interactively>(elfeed-org-export-opml record nil)
  apply(#<subr call-interactively> (elfeed-org-export-opml record nil))
  explain-pause--wrap-call-interactively(#<subr call-interactively> elfeed-org-export-opml record nil)
  apply(explain-pause--wrap-call-interactively #<subr call-interactively> (elfeed-org-export-opml record nil))
  call-interactively(elfeed-org-export-opml record nil)
  command-execute(elfeed-org-export-opml record)
  execute-extended-command(nil "elfeed-org-export-opml" nil)
  #<subr funcall-interactively>(execute-extended-command nil "elfeed-org-export-opml" nil)
  apply(#<subr funcall-interactively> (execute-extended-command nil "elfeed-org-export-opml" nil))
  funcall-interactively(execute-extended-command nil "elfeed-org-export-opml" nil)
  #<subr call-interactively>(execute-extended-command nil nil)
  apply(#<subr call-interactively> (execute-extended-command nil nil))
  explain-pause--wrap-call-interactively(#<subr call-interactively> execute-extended-command nil nil)
  apply(explain-pause--wrap-call-interactively #<subr call-interactively> (execute-extended-command nil nil))
  call-interactively(execute-extended-command nil nil)
  command-execute(execute-extended-command)
```

**Comment by LemonBreezes** [comment]
> Hi @LemonBreezes, thanks for fixing this package back to its former glory! After #81 was merged I'm getting an error for `elfeed-org-export-opml` function. Have you experienced this error as well?
> 
> Backtrace:
> 
> ```emacs-lisp
> Debugger entered--Lisp error: (wrong-type-argument fixnump t)
>   string-match("^\\[\\[\\(http.+?\\)\\]\\[\\(.+?\\)\\]\\]" "root" t)
>   #f(compiled-function (h) #<bytecode 0x10fd8ec9c4b56a3b>)((headline ... ... ... ... ... ... ...))
>   #f(compiled-function (--data) #<bytecode -0x14bda4bc676cbc52>)((headline ... ... ... ... ... ... ...))
>   mapc(#f(compiled-function (--data) #<bytecode -0x14bda4bc676cbc52>) (...))
>   #f(compiled-function (--data) #<bytecode -0x14bda4bc676cbc52>)((...))
>   org-element-ast-map((...) headline #f(compiled-function (h) #<bytecode 0x10fd8ec9c4b56a3b>) nil nil (bold citation citation-reference code entity export-snippet footnote-reference inline-babel-call inline-src-block italic line-break latex-fragment link macro ...) nil t nil)
>   org-element-map((...) headline #f(compiled-function (h) #<bytecode 0x10fd8ec9c4b56a3b>))
>   rmh-elfeed-org-convert-org-to-opml(#<buffer elfeed.org>)
>   #<subr elfeed-org-export-opml>()
>   apply(#<subr elfeed-org-export-opml> nil)
>   elfeed-org-export-opml()
>   #<subr funcall-interactively>(elfeed-org-export-opml)
>   apply(#<subr funcall-interactively> elfeed-org-export-opml)
>   funcall-interactively(elfeed-org-export-opml)
>   #<subr call-interactively>(elfeed-org-export-opml record nil)
>   apply(#<subr call-interactively> (elfeed-org-export-opml record nil))
>   explain-pause--wrap-call-interactively(#<subr call-interactively> elfeed-org-export-opml record nil)
>   apply(explain-pause--wrap-call-interactively #<subr call-interactively> (elfeed-org-export-opml record nil))
>   call-interactively(elfeed-org-export-opml record nil)
>   command-execute(elfeed-org-export-opml record)
>   execute-extended-command(nil "elfeed-org-export-opml" nil)
>   #<subr funcall-interactively>(execute-extended-command nil "elfeed-org-export-opml" nil)
>   apply(#<subr funcall-interactively> (execute-extended-command nil "elfeed-org-export-opml" nil))
>   funcall-interactively(execute-extended-command nil "elfeed-org-export-opml" nil)
>   #<subr call-interactively>(execute-extended-command nil nil)
>   apply(#<subr call-interactively> (execute-extended-command nil nil))
>   explain-pause--wrap-call-interactively(#<subr call-interactively> execute-extended-command nil nil)
>   apply(explain-pause--wrap-call-interactively #<subr call-interactively> (execute-extended-command nil nil))
>   call-interactively(execute-extended-command nil nil)
>   command-execute(execute-extended-command)
> ```

No. I'll look into it later today.

**Comment by LemonBreezes** [comment]
> Hi @LemonBreezes, thanks for fixing this package back to its former glory! After #81 was merged I'm getting an error for `elfeed-org-export-opml` function. Have you experienced this error as well?
> 
> Backtrace:
> 
> ```emacs-lisp
> Debugger entered--Lisp error: (wrong-type-argument fixnump t)
>   string-match("^\\[\\[\\(http.+?\\)\\]\\[\\(.+?\\)\\]\\]" "root" t)
>   #f(compiled-function (h) #<bytecode 0x10fd8ec9c4b56a3b>)((headline ... ... ... ... ... ... ...))
>   #f(compiled-function (--data) #<bytecode -0x14bda4bc676cbc52>)((headline ... ... ... ... ... ... ...))
>   mapc(#f(compiled-function (--data) #<bytecode -0x14bda4bc676cbc52>) (...))
>   #f(compiled-function (--data) #<bytecode -0x14bda4bc676cbc52>)((...))
>   org-element-ast-map((...) headline #f(compiled-function (h) #<bytecode 0x10fd8ec9c4b56a3b>) nil nil (bold citation citation-reference code entity export-snippet footnote-reference inline-babel-call inline-src-block italic line-break latex-fragment link macro ...) nil t nil)
>   org-element-map((...) headline #f(compiled-function (h) #<bytecode 0x10fd8ec9c4b56a3b>))
>   rmh-elfeed-org-convert-org-to-opml(#<buffer elfeed.org>)
>   #<subr elfeed-org-export-opml>()
>   apply(#<subr elfeed-org-export-opml> nil)
>   elfeed-org-export-opml()
>   #<subr funcall-interactively>(elfeed-org-export-opml)
>   apply(#<subr funcall-interactively> elfeed-org-export-opml)
>   funcall-interactively(elfeed-org-export-opml)
>   #<subr call-interactively>(elfeed-org-export-opml record nil)
>   apply(#<subr call-interactively> (elfeed-org-export-opml record nil))
>   explain-pause--wrap-call-interactively(#<subr call-interactively> elfeed-org-export-opml record nil)
>   apply(explain-pause--wrap-call-interactively #<subr call-interactively> (elfeed-org-export-opml record nil))
>   call-interactively(elfeed-org-export-opml record nil)
>   command-execute(elfeed-org-export-opml record)
>   execute-extended-command(nil "elfeed-org-export-opml" nil)
>   #<subr funcall-interactively>(execute-extended-command nil "elfeed-org-export-opml" nil)
>   apply(#<subr funcall-interactively> (execute-extended-command nil "elfeed-org-export-opml" nil))
>   funcall-interactively(execute-extended-command nil "elfeed-org-export-opml" nil)
>   #<subr call-interactively>(execute-extended-command nil nil)
>   apply(#<subr call-interactively> (execute-extended-command nil nil))
>   explain-pause--wrap-call-interactively(#<subr call-interactively> execute-extended-command nil nil)
>   apply(explain-pause--wrap-call-interactively #<subr call-interactively> (execute-extended-command nil nil))
>   call-interactively(execute-extended-command nil nil)
>   command-execute(execute-extended-command)
> ```

When you get a chance, please let me know if this alternate function definition works for you:

```
(defun rmh-elfeed-org-convert-org-to-opml (org-buffer)
  "Convert Org buffer content to OPML format.
Argument ORG-BUFFER the buffer to write the OPML content to."
  (let (need-ends
        opml-body)
    (with-current-buffer org-buffer
      (let ((org-inhibit-startup t))
        (org-mode))
      (org-element-map (rmh-elfeed-org-import-trees
                        rmh-elfeed-org-tree-id) 'headline
        (lambda (h)
          (let* ((current-level (org-element-property :level h))
                 (tags (org-element-property :tags h))
                 (heading (org-element-property :raw-value h))
                 (link-and-title (progn (string-match "^\\[\\[\\(http.+?\\)\\]\\[\\(.+?\\)\\]\\]" heading)
                                        (list (match-string-no-properties 0 heading)
                                              (match-string-no-properties 1 heading)
                                              (match-string-no-properties 2 heading))))
                 (hyperlink (progn (string-match "^\\[\\[\\(http.+?\\)\\]\\(?:\\[.+?\\]\\)?\\]" heading)
                                   (list (match-string-no-properties 0 heading)
                                         (match-string-no-properties 1 heading))))
                 url
                 title
                 opml-outline)
            ;; fill missing end outlines
            (while (and (car need-ends) (>= (car need-ends) current-level))
              (let* ((level (pop need-ends)))
                (setq opml-body (concat opml-body (format "  %s</outline>\n"
                                                          (make-string (* 2 level) ? ))))))

            (cond ((string-prefix-p "http" heading)
                   (setq url heading)
                   (setq title (or (elfeed-feed-title (elfeed-db-get-feed heading)) "Unknown")))
                  (link-and-title (setq url (nth 1 link-and-title))
                                  (setq title (nth 2 link-and-title)))
                  (hyperlink (setq url (nth 1 hyperlink))
                             (setq title (or (elfeed-feed-title (elfeed-db-get-feed (nth 1 hyperlink))) "Unknown")))
                  (t (setq title heading)))
            (if url
                (setq opml-outline (format "  %s<outline title=\"%s\" xmlUrl=\"%s\"/>\n"
                                           (make-string (* 2 current-level) ? )
                                           (xml-escape-string title)
                                           (xml-escape-string url)))
              (unless (string-prefix-p "entry-title" heading)
                (unless (member rmh-elfeed-org-tree-id tags)
                  ;; insert category title only when it is neither the top
                  ;; level elfeed node nor the entry-title node
                  (progn
                    (push current-level need-ends)
                    (setq opml-outline (format "  %s<outline title=\"%s\">\n"
                                               (make-string (* 2 current-level) ? )
                                               (xml-escape-string title)))))))
            (setq opml-body (concat opml-body opml-outline))))))

    ;; fill missing end outlines at end
    (while (car need-ends)
      (let* ((level (pop need-ends)))
        (setq opml-body (concat opml-body (format "  %s</outline>\n"
                                                  (make-string (* 2 level) ? ))))))
    opml-body))
```

**Comment by CrCs2O4** [comment]
Thanks for a quick reply! Now error messages are gone, but no new temporary buffer with OPML pops up.

I'm testing against org example from https://github.com/remyhonig/elfeed-org#solution

BTW, is it enough to put code in the source block and evaluate it or am I doing smth wrong?

```
#+begin_src elisp
(defun rmh-elfeed-org-convert-org-to-opml (org-buffer)
 ...
 opml-body))
#+end_src

#+RESULTS:
: rmh-elfeed-org-convert-org-to-opml
```

**Comment by LemonBreezes** [comment]
> Thanks for a quick reply! Now error messages are gone, but no new temporary buffer with OPML pops up.
> 
> I'm testing against org example from [remyhonig/elfeed-org#solution](https://github.com/remyhonig/elfeed-org#solution)
> 
> BTW, is it enough to put code in the source block and evaluate it or am I doing smth wrong?
> 
> ```
> #+begin_src elisp
> (defun rmh-elfeed-org-convert-org-to-opml (org-buffer)
>  ...
>  opml-body))
> #+end_src
> 
> #+RESULTS:
> : rmh-elfeed-org-convert-org-to-opml
> ```

Yes, if you evaluate an Elisp block synchronously, it gets evaluated in your current Emacs environment.

Thanks. I was able to reproduce the issue with the Org example in [remyhonig/elfeed-org#solution](https://github.com/remyhonig/elfeed-org#solution). Also, I added a significant performance optimization for how Elfeed loads the Org files.

**Comment by CrCs2O4** [comment]
thanks a lot! The pull request resolved an issue with example config and my own elfeed-org file. Well done! 👍

**Comment by remyhonig** [comment]
Thanks! I merged #88
